### PR TITLE
TS-1509: Remove TS_PARSE_OK constant

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/597/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/493/ for details.

File: lib/ts/apidefs.h.in [hunk 1]
```diff
@@ -124,8 +124,7 @@ typedef struct {
 typedef enum {
   TS_PARSE_ERROR = -1,
   TS_PARSE_DONE  = 0,
-  TS_PARSE_OK    = 1,
-  TS_PARSE_CONT  = 2,
+  TS_PARSE_CONT  = 1,
```
Comment (jpeach) [inline lib/ts/apidefs.h.in]: I think it would be kinder to plugin authors to leave ``TS_PARSE_CONT = 2`` and ``#define TS_PARSE_OK TS_PARSE_DONE``.

**Comment by jpeach** [comment]
This also needs to remove ``MIMEParseResult::PARSE_OK`` since that is where these resutn codes originate.

File: lib/ts/apidefs.h.in (hunk 1 above)
Comment (SolidWallOfCode) [inline lib/ts/apidefs.h.in]: I disagree. I think we should kill it with fire. Why pollute the namespace with a useless name?

Comment (jpeach) [inline lib/ts/apidefs.h.in]: I can live with that too, just trying to make is easier for 3rd party plugins to make the transition.

**Comment by bryancall** [comment]
@jpeach I updated `MIMEParseResult` to have the same values as TSParseResult.  I will file a bug for getting rid of `MIMEParseResult::PARSE_OK` and now that the value is only used internally it won't be an incompatible change.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/601/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/497/ for details.

**Comment by jpeach** [comment]
👍

**Pull request merged**